reV/supply_curve/supply_curve.py

```diff
@@ -554,43 +862,80 @@ def _calculate_total_lcoe_friction(self):
             logger.info('Found mean LCOE with friction. Adding key '
                         '"total_lcoe_friction" to trans table.')
 
-    def full_sort(self, trans_table=None, sort_on='total_lcoe',
-                  columns=('trans_gid', 'trans_capacity', 'trans_type',
-                           'trans_cap_cost', 'dist_mi', 'lcot', 'total_lcoe')):
+    def _competitive_wind_farms(self, wind_dirs, sc_gid, downwind=False):
         """
-        run supply curve sorting in serial
+        Exclude non-competitive wind farms for given sc_gid
 
         Parameters
         ----------
-        trans_table : pandas.DataFrame | NoneType
+        wind_dirs : CompetitiveWindFarms
+            Pre-initilized CompetitiveWindFarms instance
+        sc_gid : int
+            Supply curve gid to exclude non-competitive wind farms around
+        downwind : bool, optional
+            Flag to remove downwind neighbors as well as upwind neighbors,
+            by default False
+
+        Returns
+        -------
+        wind_dirs : CompetitiveWindFarms
+            updated CompetitiveWindFarms instance
+        """
+        gid = wind_dirs['sc_gid', sc_gid]
+        if wind_dirs.mask[gid]:
+            upwind_gids = wind_dirs['upwind', gid]
+            for n in upwind_gids:
+                check = wind_dirs.mask_gid(n)
+                if check:
+                    sc_gids = wind_dirs['sc_point_gid', n]
+                    for sc_id in sc_gids:
+                        self._mask[sc_id] = False
+
+        if downwind:
+            downwind_gids = wind_dirs['downwind', gid]
```

Comment:
You use all four of the `wind_dirs.__getitem__` options here but its not clear what is being input/output. There's so much confusion between sc_gid and sc_point_gid, and of course all the other gids (which I know is not the case here, but still). I would suggest splitting the getitem into four explicit methods.

Reply:
see above(s)